pkgs/development/python-modules/petsc4py/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv
+, fetchPypi
+, python
+, buildPythonPackage
+, cython
+, numpy
+, petsc
+, openmpi
+, openssh
+, mpi4py
+, pytest }:
+
+buildPythonPackage rec {
+  pname = "petsc4py";
+  version = "3.10.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "094hcnran0r2z1wlvmjswsz3ski1m9kqrl5l0ax8jjhnk55x0flh";
+  };
+
+  PETSC_DIR = "${petsc}";
+
+  setupPyBuildFlags = [ "build_src --force" ];
+
+  buildInputs = [
+    python
+    cython
+    numpy
+    petsc
+    openmpi
+    openssh
+    mpi4py
+  ];
```

Comment:
Some of those should probably be propagated. Why is python there?

Reply:
Hi @dotlambda, thank you for your comments. I admit that I'm not sure about buildInputs versus propagatedBuildInputs. Do you have any suggestions or guidelines for determining which packages should be propagated?

The python package is not needed, and I have removed it from the list.

Reply:
See https://nixos.org/nixpkgs/manual/#reference:

> If something is exclusively a build-time dependency, use buildInputs; if it is (also) a runtime dependency, use propagatedBuildInputs.

Reply:
See above for `cython`.
I don't find `mpi4py` in their [setup.py](https://bitbucket.org/petsc/petsc4py/src/6fe2748c55f524f51f93dc8d119d650fee44f9d0/setup.py).
`numpy` is in `install_requires` and should thus be propagated.
What do you infer the dependency on openssh from?